201-managed-application-usingmsi-for-management/azuredeploy.json

```diff
@@ -0,0 +1,97 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "allowedValues": [ "australiaeast", "eastus", "westeurope" ],
+            "metadata": {
+                "description": "Location for the resources."
+            }
+        },
+        "applicationName": {
+            "type": "string",
+            "metadata": {
+                "description": "Managed application name"
+            },
+            "defaultValue": "MsiManagedApplication"
+        },
+        "logicAppName": {
+            "type": "string",
+            "defaultValue": "logicappwfsdc"
+        },
+        "rolenameguid": {
+            "type": "string"
+        },
+        "storageAccountName": {
+            "type": "string",
+            "defaultValue": "containerstor"
+        },
+        "_artifactsLocation": {
+            "type": "string",
+            "metadata": {
+                "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+            },
+            "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-managed-application-usingmsi-for-management/"
+        },
+        "_artifactsLocationSasToken": {
+            "type": "securestring",
+            "metadata": {
+                "description": "The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated."
+            },
+            "defaultValue": ""
+        }
+    },
+    "variables": {
+        "lockLevel": "None",
+        "description": "Sample Managed application definition",
+        "displayName": "Sample Managed application definition",
+        "managedApplicationDefinitionName": "[concat(parameters('applicationName'), '_ApplicationDefinition')]",
+        "packageFileUri": "[uri(parameters('_artifactsLocation'), concat('artifacts/ManagedAppZip/pkg.zip', parameters('_artifactsLocationSasToken')))]",
+        "managedResourceGroupId": "[concat(subscription().id,'/resourceGroups/', concat(parameters('applicationName'),'_managed'))]",
```

Comment:
remove concat() and use resourceId()

Reply:
This is a resource group id. I could not find how to use a function to create this in a template

Reply:
use subscriptionResourceId()

Reply:
That does not work either cause that function needs resource type. I tried doing :
"managedResourceGroupId": "[subscriptionResourceId('resourceGroups', concat(parameters('applicationName'),'_managed'))]",

But that fails with error :

Unable to evaluate template language function 'subscriptionResourceId': function requires exactly one multi-segmented argument which must be resource type including resource
provider namespace